### PR TITLE
K8SPSMDB-451: Take mongos into consideration for CR status

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPSMDB-451](https://jira.percona.com/browse/K8SPSMDB-451) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: pkg/controller/perconaservermongodb/status.go [hunk 1]
```diff
@@ -173,7 +173,11 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 	case replsetsPaused == len(repls):
 		cr.Status.State = api.AppStatePaused
 	case !inProgress && replsetsReady == len(repls) && clusterState == api.AppStateReady && cr.Status.Host != "":
-		cr.Status.State = api.AppStateReady
+		if cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status != api.AppStateReady {
+			cr.Status.State = cr.Status.Mongos.Status
+		} else {
```
Comment (qJkee) [inline pkg/controller/perconaservermongodb/status.go:178]: we don't need else if we keep old 176 line

Comment (egegunes) [reply pkg/controller/perconaservermongodb/status.go]: fixed: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/87f2f517688eac8dd529eebdd582b9512a71bada

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| upgrade | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| liveness | passed |
| smart-update | passed |
| pitr | passed |
| version-service | passed |
| users | passed |
| pitr-sharded | passed |
| data-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/0818bf2bc25fca15bb8ab8ec629103d210b0b8a2
image: `perconalab/percona-server-mongodb-operator:PR-696-0818bf2b`

**Review by dAdAbird — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| operator-self-healing | passed |
| liveness | passed |
| upgrade | passed |
| operator-self-healing-chaos | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/87f2f517688eac8dd529eebdd582b9512a71bada
image: `perconalab/percona-server-mongodb-operator:PR-696-87f2f517`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| pitr | passed |
| users | failed |
| data-sharded | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/e804ca1b20e84ff9adf19961cf3b29bd10853a97
image: `perconalab/percona-server-mongodb-operator:PR-696-e804ca1b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/696/commits/88daff3a20a407b7789c8a37efdde57aed0646b5
image: `perconalab/percona-server-mongodb-operator:PR-696-88daff3a`

**Pull request merged**